### ISSUE TITLE
Incoming spatial profile messages need to be added to the existing spatial profile cache

### ISSUE THREAD

**Issue by veggiesaurus** [CARTAvis/carta-frontend#386]
Related to [carta-backend/273](https://github.com/CARTAvis/carta-backend/issues/273). Once this code is merged into the backend, it will only send _required_ new spatial profiles after a `SET_SPATIAL_REQUIREMENTS` message is received. For example, if the requirements were `["x"]` and change to `["x", "y"]`, only the y-profile will be sent, as the x-profile is assumed to be present. 

The frontend needs to add the new profile to the existing ones, rather than clearing out the old profiles and replacing then with the new profile
